contactform/ContactFormPlugin.php

```diff
@@ -79,27 +79,70 @@ public function getSettingsHtml()
 	 */
 	public function setSettings($values)
 	{
-		if (!$values)
+    $settings = $this->prepSettings($values);
+
+    // Craft::dd(craft()->request->getPost('settings'));
+
+		parent::setSettings($settings);
+	}
+
+  /**
+	 * @return array
+	 */
+  public function prepSettings($values) {
+
+    if (!$values)
 		{
 			$values = array();
 		}
 
 		if (is_array($values))
 		{
+      $postedSettings = craft()->request->getPost('settings');
```

Comment:
These setSettings/prepSettings changes seem unrelated to the mailing list feature.  Can you explain what's going on? Why pull settings from POST here now?